os/include/tinyara/sched.h

```diff
@@ -86,18 +86,24 @@
 /********************************************************************************
  * Pre-processor Definitions
  ********************************************************************************/
-#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(CONFIG_ARM_MPU)
+#if defined(CONFIG_ARM_MPU) && (defined(CONFIG_APP_BINARY_SEPARATION) || defined(CONFIG_MPU_STACK_OVERFLOW_PROTECTION))
 enum {
 	REG_RNR,
 	REG_RBAR,
-	REG_RASR
+	REG_RASR,
+	REG_MAX
 };
 
+#if defined(CONFIG_MPU_STACK_OVERFLOW_PROTECTION)
+#define STACKGUARD_MPU_NUM		3	/* Region number 3 is reserved for stack protection */
```

Comment:
STACKGUARD_MPU_REGION_NUM would be good.

Reply:
yes changed